app/gui2/src/components/GraphEditor/widgets/WidgetFunction.vue

```diff
@@ -306,9 +317,12 @@ export const widgetDefinition = defineWidget(WidgetInput.isFunctionCall, {
     // and to resolve the infix call as its own application.
     if (prevFunctionState?.callId === ast.id) return Score.Mismatch
 
+    // We only render the function widget on the top-level AST node.
```

Comment:
This comment may suggest we display it only on node root, but actually this is an _application chain's_ top level.

Reply:
Adjusted